updater.go

```diff
@@ -263,6 +332,65 @@ func (u *Updater) NeedUpdate(ctx Context) (upToDate bool, err error) {
 	return update.NeedUpdate, nil
 }
 
+func (u *Updater) CheckAndDownload(ctx Context) (updateAvailable, updateCached bool, err error) {
+	options := ctx.UpdateOptions()
+	update, err := u.checkForUpdate(ctx, options)
+	if err != nil {
+		return false, false, err
+	}
+
+	if !update.NeedUpdate {
+		return false, false, nil
+	}
+
+	if u.updateMissingAsset(update) {
+		return false, false, nil
+	}
+
+	var tmpDir string
+	defer func() {
+		// If anything in this process errors cleanup the downloaded asset
+		if err != nil {
+			if err := u.CleanupPreviousUpdates(); err != nil {
+				u.log.Infof("Error cleaning up previous downloads: %v", err)
+			}
+		}
+		if tmpDir != "" {
+			u.Cleanup(tmpDir)
+		}
+	}()
+	var digestChecked bool
+	updateAvailable = true
+	downloadedAssetPath, err := u.FindDownloadedAsset(update.Asset.Name)
+	if downloadedAssetPath == "" || err != nil {
+		u.log.Infof("Could not find existing download asset for version: %s. Downloading new asset.", update.Version)
+		tmpDir = u.tempDir()
+		// This will set update.Asset.LocalPath
+		if err := u.downloadAsset(update.Asset, tmpDir, options); err != nil {
+			return false, false, downloadErr(err)
+		}
+		updateCached = true
+		digestChecked = true
+		downloadedAssetPath = update.Asset.LocalPath
+	}
+	updateCached = true
```

Comment:
i don't understand the purpose of this var anymore. it is equivalent to `updateAvailable`, as both will always be true if or false now. The original purpose was `updateDownloaded` since we wanted a flag for tests, why the change?

Reply:
@joshblum `download-latest` should only return `true` to the service if an update is available via the API server and it's either:

1. Already on disk
2. We successfully downloaded it to disk.

Need both so that if a download fails, we don't return `true` to the service and therefore to the GUI. The GUI should only display an update if the download has been cached.

The original signature of this function had `availableAndDownloaded` as the named return value. It was split so that unit tests could check with more granularity.